src/components/frontend-engine/frontend-engine.tsx

```diff
@@ -74,8 +112,25 @@ const FrontendEngineInner = forwardRef<IFrontendEngineRef, IFrontendEngineProps>
 
 			return () => subscription.unsubscribe();
 		}
+
+		// eslint-disable-next-line react-hooks/exhaustive-deps
 	}, [checkIsFormValid, onChange, watch]);
 
+	useDeepCompareEffectNoCheck(() => {
+		const apiErrors = Object.fromEntries(Object.entries(errors).filter(([_, value]) => value.type === "api"));
+
+		if (apiErrors && !isEmpty(apiErrors) && oldFormValues) {
+			Object.keys(apiErrors).forEach((key) => {
+				const oldValue = oldFormValues[key];
+				const updatedValue = formValues[key];
+
+				if (oldValue !== updatedValue) {
+					clearErrors(key);
+				}
+			});
+		}
+	}, [formValues, oldFormValues]);
```

Comment:
can remove `oldFormValues` as a dependency for 2 reasons:
- `oldFormValues` is updated when `formValues` change so there is no point checking `oldFormValues`
- `oldFormValues` is a `ref` which does not trigger useEffect